test/dest-symlinks.js

```diff
@@ -94,7 +189,8 @@ describe('.dest() with symlinks', function() {
       path: inputDirpath,
       contents: null,
       stat: {
-        isDirectory: isDirectory,
+        isSymbolicLink: isSymbolicLink(true),
+        isDirectory: isDirectory(true),
```

Comment:
This can never happen

Reply:
That's what I thought until you suggested we should handle users faking stuff like this :-) Did I misunderstand you (way) earlier?

Reply:
From above:
> I've mostly given up on the use-case of synthetically-generated, incorrect Vinyl objects for this implementation. I believe we can solve that desired use-case on the next iteration of this concept (when we have better-stats)